.travis.yml

```diff
@@ -10,81 +10,75 @@ cache:
     # compiler calls. It includes a few flags that aren't used by one half of
     # the process or the other. -Qunused-arguments prevents clang from failing
     # the build because of this.
-    - export CXX="ccache clang++ -Qunused-arguments" CC="ccache clang -Qunused-arguments"
-    - if [[ $BOND_LANG == *gcc ]]; then export CXX="ccache g++-4.8" CC="ccache gcc-4.8"; fi
-    # linux prereqisite packages
-    - if [ "$TRAVIS_OS_NAME" == "linux" ]; then wget --no-check-certificate https://www.cmake.org/files/v3.2/cmake-3.2.3-Linux-x86_64.tar.gz; fi
-    - if [ "$TRAVIS_OS_NAME" == "linux" ]; then tar -xzvf cmake-3.2.3-Linux-x86_64.tar.gz; fi
-    - if [ "$TRAVIS_OS_NAME" == "linux" ]; then export PATH=$PWD/cmake-3.2.3-Linux-x86_64/bin:$PATH; fi
+    - export CXX="ccache $CXX -Qunused-arguments --system-header-prefix=boost/"
+    - export CC="ccache $CC -Qunused-arguments --system-header-prefix=boost/"
+    - echo "CXX=$CXX"
+    - echo "CC=$CC"
     # nunit installation
     - if [ "$BOND_LANG" == "cs" ]; then travis_retry nuget install NUnit.Runners -version 2.6.4; fi
-    - export PATH=/opt/cabal/1.22/bin:/opt/ghc/${BOND_GHC_VERSION#?}/bin:$PATH
+    - export PATH=/opt/cabal/1.22/bin:/opt/ghc/${GHC_VERSION}/bin:$PATH
     - cabal update
-    - cabal install happy
+    - cabal install Cabal happy
 
 before_script:
     # restore nuget packages for solution
     - if [ "$BOND_LANG" == "cs" ]; then travis_retry nuget restore cs/cs.sln; fi
 
 script:
     - mkdir build && cd build
-    - cmake ..
+
+    - cmake -DBOND_CORE_ONLY:bool=$CORE_ONLY ..
     - if [ "$BOND_LANG" == "cs" ]; then make --jobs 2 DESTDIR=$HOME install; fi
+    - if [ "$BOND_LANG" == "cs" ]; then cd ..; fi
     - if [ "$BOND_LANG" == "cs" ]; then export BOND_COMPILER_PATH=$HOME/usr/local/bin; fi
-    - if [[ $BOND_LANG == cpp* ]]; then make --jobs 2 check; fi
-    - if [ "$TRAVIS_OS_NAME" == "linux" ]; then export BOND_PYTHON_TARGETS="python_unit_test python_extension" BOND_PYTHON_TESTS=python_unit_test; fi
-    - if [[ $BOND_LANG == py* ]]; then make $BOND_PYTHON_TARGETS; fi
-    - if [[ $BOND_LANG == py* ]]; then ctest --tests-regex $BOND_PYTHON_TESTS --output-on-failure; fi
-    - if [ "$BOND_GHC_VERSION" != "" ]; then make gbc-tests; fi
-    - cd ..
     - if [ "$BOND_LANG" == "cs" ]; then xbuild /p:Configuration=Debug cs/cs.sln; fi
     - if [ "$BOND_LANG" == "cs" ]; then xbuild /p:Configuration=Fields cs/cs.sln; fi
     - if [ "$BOND_LANG" == "cs" ]; then mono NUnit.Runners.2.6.4/tools/nunit-console.exe -framework=mono-4.5 -labels cs/test/core/bin/debug/net45/Properties/Bond.UnitTest.dll cs/test/core/bin/debug/net45/Fields/Bond.UnitTest.dll cs/test/internal/bin/debug/net45/Bond.InternalTest.dll; fi
-    - if [ "$BOND_GHC_VERSION" != "" ]; then cd compiler; fi
-    - if [ "$BOND_GHC_VERSION" != "" ]; then ../build/compiler/build/gbc-tests/gbc-tests; fi
-
-before_cache:
-    - if [ "$BOND_GHC_VERSION" != "" ]; then rm -rf compiler/.cabal-sandbox; fi
```

Comment:
Did you intend to remove this `before_cache` section? (It looks like a `cd ..` is missing anyway...)